docs: Add SDK Search reference (#1065)
vtex/faststore: docs/docs/reference/sdk/search/usePagination.mdx

import PropsComponent from "@site/src/components/PropsComponent/PropsComponent";

# usePagination

The `usePagination` hook provides the pagination links (i.e., `next`/`prev`) of a search result. You can use the `rel="next"` and `rel="prev"` links along with the `<link />` tag to add navigation buttons to your search result pages.

## Import

```tsx
import { usePagination } from '@faststore/sdk'
```

## Usage

```tsx
function Component () {
  const totalProducts = 100
  const { next, prev } = usePagination(totalProducts)

  return (
    <>
      {prev && <a href={prev.link}>Previous Page</a>}

>       {next && <a href={next.link}>Next Page</a>}

Reviewer: Recommend adding key props to the example.

Author: key? for what?

import PropsComponent from "@site/src/components/PropsComponent/PropsComponent";

# usePagination

> This hooks gives you the links to next/prev pages on the search pagination.
> This is helpfull for creating pagination markers like `<link rel="next"/>` and `<link rel="prev"/>` tags and buttons in the UI

Reviewer 2: ```suggestion
The `usePagination` hook provides the pagination links (i.e., `next`/`prev`) of a search result. You can use the `rel="next"` and `rel="prev"` links along with the `<link />` tag to add navigation buttons to your search result pages.
```

## Import

```tsx
import { usePagination } from '@faststore/sdk'
```

## Usage

```tsx
function Component () {
  const totalProducts = 100
  const { next, prev } = usePagination(totalProducts)

  return (
    <>
      {prev && <a href={prev.link}>Previous Page</a>}
      {next && <a href={next.link}>Next Page</a>}
    </>
  )
}
```

## Props
<PropsComponent
  type="number"

>   description="Total number of products in the search result"

Reviewer 2: ```suggestion
  description="Total number of products returned by the search query."
```

/>

## Return
<PropsComponent
    name="prev"
    type="{ cursor: number; link: string} | false"

>     description="Link and cursor for previous page on search pagination"

Reviewer 2: ```suggestion
    description="Cursor position and corresponding link for the previous page of the search result."
```

/>

<PropsComponent
    name="next"
    type="{ cursor: number; link: string} | false"

>     description="Link and cursor for next page on search pagination"

Reviewer 2: ```suggestion
    description="Cursor position and corresponding link for the next page of the search result."
```
